### PR TITLE
feat(django): add doc on automatic db migrations for django

### LINKED ISSUE #1964: Add doc on Django on how to do a database migration with the post deploy hook
https://doc.scalingo.com/platform/app/postdeploy-hook

https://doc.scalingo.com/languages/python/django/start#create-your-application-and-databases-on-scalingo

### PR DESCRIPTION
**Pull request by yohann-bacha**
Fixes #1964

### CODE REVIEW

File: src/_posts/languages/python/django/2000-01-01-start.md [hunk 1]
```diff
@@ -233,6 +233,29 @@ Then change the environment variable of your application with a coma-separated l
 scalingo -a app-name env-set ALLOWED_HOSTS=app-name.osc-fr1.scalingo.io,example.com
 ```
 
+### Automatic database migrations
```
Comment (EtienneM) [inline src/_posts/languages/python/django/2000-01-01-start.md:236]: Titles in English:

```suggestion
### Automatic Database Migrations
```

File: src/_posts/languages/python/django/2000-01-01-start.md [hunk 2]
```diff
@@ -233,6 +233,29 @@ Then change the environment variable of your application with a coma-separated l
...
+
+If your Django application is related to a database, you can set it up to deploy automatically after a deployment, 
+thanks to the [postdeploy hook]({% post_url platform/app/postdeploy-hook %}).
```
Comment (EtienneM) [inline src/_posts/languages/python/django/2000-01-01-start.md:239]: I don't understand what "deploying automatically" a database means. You may have used the wrong word?

File: src/_posts/languages/python/django/2000-01-01-start.md [hunk 3]
```diff
@@ -233,6 +233,29 @@ Then change the environment variable of your application with a coma-separated l
...
+
+Database migrations in Django are separated in two steps :
```
Comment (EtienneM) [inline src/_posts/languages/python/django/2000-01-01-start.md:241]: no space in English:

```suggestion
Database migrations in Django are separated in two steps:
```

File: src/_posts/languages/python/django/2000-01-01-start.md [hunk 4]
```diff
@@ -233,6 +233,29 @@ Then change the environment variable of your application with a coma-separated l
...
+
+- `python manage.py makemigrations`, which creates migrations files from model change
+- `python manage.py migrate`, which applies the migration file to the related database
+
+Because `makemigrations` modifies the source code, this command must be done on your computer, and the created
+migration files committed to Git. This is not a command to execute on your postdeploy hook.
```
Comment (EtienneM) [inline src/_posts/languages/python/django/2000-01-01-start.md:247]: not sure butI think it's more correct:

```suggestion
migration files committed to Git. This is not a command to execute in your postdeploy hook.
```

File: src/_posts/languages/python/django/2000-01-01-start.md [hunk 5]
```diff
@@ -233,6 +233,29 @@ Then change the environment variable of your application with a coma-separated l
...
+
+The `migrate` command is the one to execute to apply to migrations to your database. You can add it to your `Procfile`
+like this :
```
Comment (EtienneM) [inline src/_posts/languages/python/django/2000-01-01-start.md:250]:
```suggestion
The `migrate` command is the one to execute to apply to migrations the your database. You can add it to your `Procfile`
like this:
```

**Review by EtienneM — CHANGES_REQUESTED** [review]

File: src/_posts/languages/python/django/2000-01-01-start.md (hunk 2 above)
Comment (yohann-bacha) [reply src/_posts/languages/python/django/2000-01-01-start.md:239]: This is totally bad worded, I will check that

File: src/_posts/languages/python/django/2000-01-01-start.md (hunk 1 above)
Comment (yohann-bacha) [reply src/_posts/languages/python/django/2000-01-01-start.md:236]: Yup, I'll do this

File: src/_posts/languages/python/django/2000-01-01-start.md (hunk 3 above)
Comment (yohann-bacha) [reply src/_posts/languages/python/django/2000-01-01-start.md:241]: i'll never get those

File: src/_posts/languages/python/django/2000-01-01-start.md (hunk 4 above)
Comment (yohann-bacha) [reply src/_posts/languages/python/django/2000-01-01-start.md:247]: Well why not, I'll include them

File: src/_posts/languages/python/django/2000-01-01-start.md [hunk 6]
```diff
@@ -233,6 +233,29 @@ Then change the environment variable of your application with a coma-separated l
 scalingo -a app-name env-set ALLOWED_HOSTS=app-name.osc-fr1.scalingo.io,example.com
 ```
 
+### Automatic Database Migrations
+
+If your Django application is related to a database, you can set it up to migrate your database schema automatically 
+after a deployment, thanks to the [postdeploy hook]({% post_url platform/app/postdeploy-hook %}).
```
Comment (EtienneM) [inline src/_posts/languages/python/django/2000-01-01-start.md:239]: question: sorry I didn't noticed it at the first review. But I think that having a database is mandatory for a Django application. Hence I'm not sure the beginning of this sentence is mandatory.
```suggestion
You can configure a [postdeploy hook]({% post_url platform/app/postdeploy-hook %}) to migrate your database schema automatically after a deployment.
```

**Review by EtienneM — COMMENTED** [review]
Just a question, then LGTM

**Review by EtienneM — APPROVED** [review]

**Review by EtienneM — CHANGES_REQUESTED** [review]
It seems that you have a problem in your `post_url` syntax. Check the review app deployment logs.

**Review by EtienneM — APPROVED** [review]

**Pull request merged**